### PR TITLE
Each rollup references its batches, rather than a set of transactions

### PR DESCRIPTION
**Pull request by joeldudley**
### Why is this change needed?

Rollups should point to the batches they are created from. They no longer need to contain the transaction information, since that is stored in the batch.

Each rollup only references one batch for now, as the mapping between rollups and batches is one-to-one. In the future, a single rollup will reference multiple batches.

### What changes were made as part of this PR:

- Provide a high level list of the changes made

### Definition of done

- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated 
- [ ] End-to-end tests run if required (see below)

### End-to-end tests

Running the end-to-end tests in the [obscuro-test repo](https://github.com/obscuronet/obscuro-test) is currently at the 
discretion of the developer prior to merging a PR. The intention is not to slow down development by having the longer 
running end-to-end tests as a PR gate run on each branch commit etc. To manually trigger a run pre-merge;

- Go to [run_local_tests](https://github.com/obscuronet/obscuro-test/actions/workflows/run_local_tests.yml)
- Click the "Run workflow" button
- Use the main branch of obscuro-test 
- Enter the name of the PR branch for go-obscuro
- Run the workflow

The run takes approximately 20 minutes and any failure will be reported in the workflow output. Should the tests fail 
the docker container output for the local testnet and all test artifacts will be added to the run and can be downloaded
for debugging and analysis, with a retention period of 2 days. 

Note that every PR merge to main will also trigger a run of post-merge tests, so even if you do not manually trigger 
pre-merge, tests will be run post-merge. The intention being to catch any issues fast on main to allow for quick 
resolution. The post-merge test output can be seen at 
[run_merge_tests](https://github.com/obscuronet/obscuro-test/actions/workflows/run_merge_tests.yml) and will show both 
the PR number and author in the list of runs.

### CODE REVIEW

File: go/common/headers.go [hunk 1]
```diff
@@ -154,36 +154,6 @@ func (r *RollupHeader) Hash() L2RootHash {
 	return hash
 }
 
-func (r *RollupHeader) ToBatchHeader() *BatchHeader {
```
Comment (joeldudley) [inline go/common/headers.go:157]: This was only used for some unit tests, so I removed it.

**Review by BedrockSquirrel — COMMENTED** [review]
I'm a bit confused here, I thought we needed to be able to go from rollup -> full batches (not by lookup but the full data (compressed)).

So by my understanding, ExtRollup can't just have batch hashes on it, it needs the full tx data of each batch. I may have just missed something, or maybe this is a stepping stone towards that somehow (bit odd if it were though because it had the tx data before).

**Comment by joeldudley** [comment]
I'm not sure. I thought the rollups were just a commitment to the batches distributed by the sequencer.

@tudor-malene - wdyt? Do we need the full compressed batches in the rollup, or just the hashes of the batches we're committing to?

**Comment by joeldudley** [comment]
@tudor-malene Ok thanks, will rework.

File: go/enclave/core/rollup.go [hunk 2]
```diff
@@ -44,29 +38,20 @@ func (r *Rollup) IsGenesis() bool {
...
-		Header:       encryptedRollup.Header,
-		Transactions: transactionBlobCrypto.Decrypt(encryptedRollup.EncryptedTxBlob),
-	}
-}
-
-func (r *Rollup) ToBatch() *Batch {
-	return &Batch{
-		Header:       r.Header.ToBatchHeader(),
-		Transactions: r.Transactions,
+		Header: encryptedRollup.Header,
```
Comment (BedrockSquirrel) [inline go/enclave/core/rollup.go:55]: Should this populate the batches here?

**Review by BedrockSquirrel — COMMENTED** [review]
LGTM, small thing I think was missed

**Review by BedrockSquirrel — APPROVED** [review]
LGTM, thanks

**Pull request merged**